support-frontend/assets/helpers/abTests/__tests__/abtestTest.ts

```diff
@@ -21,840 +18,303 @@ jest.mock('ophan', () => ({
 }));
 
 // ----- Tests ----- //
-const emptySettings = {
-	switches: {
-		experiments: {},
-	},
-	amounts: {
-		GBPCountries: {},
-		UnitedStates: {},
-	},
-};
-
-const acquisitionDataMockTestControl = {
-	componentId: 'acquisitions-subscription-banner',
-	componentType: 'ACQUISITIONS_SUBSCRIPTIONS_BANNER',
-	abTest: {
-		name: 'mockTest',
-		variant: 'control',
-	},
-};
-
-const acquisitionDataMockTestVariant = {
-	componentId: 'acquisitions-subscription-banner',
-	componentType: 'ACQUISITIONS_SUBSCRIPTIONS_BANNER',
-	abTest: {
-		name: 'mockTest',
-		variant: 'variant',
-	},
-};
-
-const acquisitionDataMockTest2Control = {
-	componentId: 'acquisitions-subscription-banner',
-	componentType: 'ACQUISITIONS_SUBSCRIPTIONS_BANNER',
-	abTest: {
-		name: 'mockTest2',
-		variant: 'control',
-	},
-};
-
-const acquisitionDataMockTest2Variant = {
-	componentId: 'acquisitions-subscription-banner',
-	componentType: 'ACQUISITIONS_SUBSCRIPTIONS_BANNER',
-	abTest: {
-		name: 'mockTest2',
-		variant: 'variant',
-	},
-};
-
-const mockTestControl = `/test.html?acquisitionData=${encodeURI(
-	JSON.stringify(acquisitionDataMockTestControl),
-)}`;
-
-const mockTestVariant = `/test.html?acquisitionData=${encodeURI(
-	JSON.stringify(acquisitionDataMockTestVariant),
-)}`;
-
-const mockTest2Control = `/test.html?acquisitionData=${encodeURI(
-	JSON.stringify(acquisitionDataMockTest2Control),
-)}`;
-
-const mockTest2Variant = `/test.html?acquisitionData=${encodeURI(
-	JSON.stringify(acquisitionDataMockTest2Variant),
-)}`;
-
-describe('basic behaviour of init', () => {
+
+describe('init', () => {
 	Object.defineProperty(window, 'matchMedia', {
-		value: jest.fn().mockImplementation(() => ({
+		value: jest.fn().mockReturnValue({
 			matches: false,
-		})),
+		}),
 	});
+
+	// Common arguments to init
+	const mvt = 123456;
+	const country = 'GB';
+	const countryGroupId = GBPCountries;
+	const emptySettings = {
+		switches: {
+			experiments: {},
+		},
+		amounts: {
+			GBPCountries: {},
+			UnitedStates: {},
+		},
+	} as Settings;
+
 	afterEach(() => {
 		window.localStorage.clear();
 	});
-	it('The referrerControlled user should be allocated to the control bucket', () => {
-		document.cookie = 'GU_mvt_id=12346';
-		window.history.pushState({}, 'Test Title', mockTestControl);
+
+	it('assigns a user to a variant', () => {
```

Comment:
I changed the names of all of the tests so that they read like prose with the `it` function name.

I've also removed test cases that just changed the `seed` to test assignment to `control`/`variant`. I think they're too subtle and what they're actually testing is that for a given combination of test name, `mvt`, and `seed` you're assigned to a specific variant. I think we get more value out of tests  by testing deterministic behaviour. This means most of the remaining tests are testing the targeting is working correctly.